### DISCUSSION TITLE
The requested module 'my-package' does not provide an export named

### DISCUSSION THREAD

**Discussion by Dailyscat** [webpack/webpack#13607] [General]
Good morning. First thanks for the help.
My situation is as below.
I tried to export {alert, confirm} in the original index.js file and use it through import {alert, confirm} from "my-package" in a file that uses that package.
However, after bundling, when I download the package and try to use the package through import {alert, confirm} from "my-package", an error like "The requested module 'my-package' does not provide an export named" is occurring. .

[my repo ](https://github.com/Dailyscat/lcp-common-window-dialog)

webpack.config.js

```
const path = require("path");
const TerserPlugin = require("terser-webpack-plugin");

module.exports = {
  entry: "./src/index.js",
  output: {
    path: path.resolve(__dirname, "dist/"),
    filename: "bundle.js",
    library: {
      type: "umd",
    },
  },
  resolve: {
    extensions: [".ts", ".js"],
  },
  module: {
    rules: [
      {
        loader: "babel-loader",
        test: /\.(js|jsx|tsx|ts)$/,
        include: [path.resolve(__dirname, "src/")],
        exclude: /node_modules/,
        options: {
          presets: [
            [
              "@babel/preset-env",
              {
                targets: { ie: 11 },
              },
            ],
            "@babel/preset-typescript",
          ],
          plugins: [["@babel/plugin-transform-runtime", { corejs: 3 }]],
        },
      },
    ],
  },

  optimization: {
    minimize: true,
    // minimizer: [
    //   new TerserPlugin({
    //     extractComments: false,
    //   }),
    // ],
  },
};
```

**Comment by alexander-akait** [comment]
`type: "umd"` is not ES modules, so you have only default export

**Reply by Dailyscat** [reply]
Even if I set the Type field as the default value and build, the same error occurs. Do you have any other opinions?

**Reply by alexander-akait** [reply]
https://webpack.js.org/configuration/output/#outputmodule

**Reply by Dailyscat** [reply]
I'm really sorry to keep bothering you.
As shown below, in index.js, in the export {function1, function2} , I gave module: true inside webpack.config.js as you said above. But it doesn't run and I keep getting the error "bundle.js' does not provide an export named".

![image](https://user-images.githubusercontent.com/35999704/122862230-c6756880-d35b-11eb-8970-2c8b7da244a8.png)

**Reply by alexander-akait** [reply]
Please provide full steps to reproduce

**Reply by Dailyscat** [reply]
First my webpack.config.js file looks like below.

```
const path = require("path");
const TerserPlugin = require("terser-webpack-plugin");

module.exports = {
  entry: "./src/index.js",
  output: {
    path: path.resolve(__dirname, "dist/"),
    filename: "bundle.js",
    module: true,
  },
  resolve: {
    extensions: [".ts", ".js"],
  },
  experiments: {
    outputModule: true,
  },
  module: {
    rules: [
      {
        loader: "babel-loader",
        test: /\.(js|jsx|tsx|ts)$/,
        include: [path.resolve(__dirname, "src/")],
        exclude: /node_modules/,
        options: {
          presets: [
            [
              "@babel/preset-env",
              {
                targets: { ie: 11 },
              },
            ],
            "@babel/preset-typescript",
          ],
          plugins: [["@babel/plugin-transform-runtime", { corejs: 3 }]],
        },
      },
    ],
  },

  optimization: {
    minimize: true,
    // minimizer: [
    //   new TerserPlugin({
    //     extractComments: false,
    //   }),
    // ],
  },
};


```

Second, the format of the index.js file I wrote is as follows.

```
const alert = dialogController.alert.bind(dialogController);
const confirm = dialogController.confirm.bind(dialogController);

...

export { alert, confirm };
```


As you can see, two functions are exported through the export {alert, confirm} format below.

Then, in bundle.js, the bundled file through this format,

I think `import {alert, confirm} from "bundle.js"` should work properly.

as i said above
I am getting this error `The requested module 'my-package' does not provide an export named`.

**Reply by alexander-akait** [reply]
Sorry, I don't know your code, so I can't say why it was not found, will be great if your provide simple github repo, yes, it should work without any problems

**Reply by Dailyscat** [reply]
thanks! greate help!

**Reply by alexander-akait** [reply]
I ask you provide small example, without it I can't ask what is wrong

**Reply by alexander-akait** [reply]
But I think you want to create ES modules library, if yes we already working on it https://github.com/webpack/webpack/pull/13622
